file.go

```diff
@@ -7,56 +7,109 @@ import (
 )
 
 var (
-	ErrNotDirectory = errors.New("Couldn't call NextFile(), this isn't a directory")
-	ErrNotReader    = errors.New("This file is a directory, can't use Reader functions")
+	ErrNotDirectory = errors.New("file isn't a directory")
+	ErrNotReader    = errors.New("file isn't a regular file")
+
+	ErrNotSupported = errors.New("operation not supported")
 )
 
 // File is an interface that provides functionality for handling
 // files/directories as values that can be supplied to commands. For
-// directories, child files are accessed serially by calling `NextFile()`.
+// directories, child files are accessed serially by calling `NextFile()`
+//
+// Read/Seek methods are only valid for files
+// NextFile method is only valid for directories
 type File interface {
-	// Files implement ReadCloser, but can only be read from or closed if
-	// they are not directories
-	io.ReadCloser
-
-	// FileName returns a filename associated with this file
-	FileName() string
-
-	// FullPath returns the full path used when adding with this file
-	FullPath() string
-
-	// IsDirectory returns true if the File is a directory (and therefore
-	// supports calling `NextFile`) and false if the File is a normal file
-	// (and therefor supports calling `Read` and `Close`)
-	IsDirectory() bool
-
-	// NextFile returns the next child file available (if the File is a
-	// directory). It will return (nil, io.EOF) if no more files are
-	// available. If the file is a regular file (not a directory), NextFile
-	// will return a non-nil error.
-	NextFile() (File, error)
+	io.Closer
+
+	// Size returns size of this file (if this file is a directory, total size of
+	// all files stored in the tree should be returned). Some implementations may
+	// choose not to implement this
+	Size() (int64, error)
```

Comment:
`Size` has always been a sticky issue for us. Do we *really* need it here or can it go on files only?

Reply:
We use it on directorise for progress bar in add

Reply:
I see. Yeah, I really can't see a better way to do this. As long as `Size` is clearly defined.